sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorEventSource.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.Tracing;
+
+namespace Azure.Messaging.EventHubs.Processor.Diagnostics
+{
+    /// <summary>
```

Comment:
Context is in the comments below, but I will say that if we decide to leave this as unsealed and virtual, we should take advantage by including tests in the same manner as the Blobs Storage Manager, at least where the important background operations are concerned.